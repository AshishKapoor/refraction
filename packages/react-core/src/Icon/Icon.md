With the `Icon` component you can insert one of the supported icons
in your component:

```js
<Icon name="globe" style={{ fontSize: '2em' }} />
```

This is the complete list of available icons:

```js noeditor
const iconNames = require('./icons/quid-icons.json').icons.map(
  ({ icon }) => icon.tags[0]
);
const cell = {
  maxWidth: 0,
};
const icon = {
  padding: '.5em 1em',
  fontSize: '2em',
};
const name = {
  whiteSpace: 'normal',
  wordWrap: 'break-word',
  minWidth: 0,
  marginLeft: '1em',
};

<div style={{ overflow: 'auto', maxHeight: 400 }}>
  <table className="DemoTable">
    <tbody>
      {iconNames
        .reduce((acc, icon, i) => {
          i % 2 === 1 &&
            acc.push([
              icon.split(',')[0],
              iconNames[i + 1] && iconNames[i + 1].split(',')[0],
            ]);
          return acc;
        }, [])
        .map((icons, i) => (
          <tr key={i}>
            <td style={cell}>
              <div style={{ display: 'flex', alignItems: 'center' }}>
                <Icon style={icon} name={icons[0]} />
                <div style={name}>{icons[0]}</div>
              </div>
            </td>

            <td style={cell}>
              <div style={{ display: 'flex', alignItems: 'center' }}>
                <Icon style={icon} name={icons[1]} />
                <div style={name}>{icons[1]}</div>
              </div>
            </td>
          </tr>
        ))}
    </tbody>
  </table>
</div>;
```
