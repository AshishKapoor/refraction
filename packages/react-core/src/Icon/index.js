// @flow
import * as React from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled/macro';

type Props = {
  /** Icon name */
  name: string,
  /** Forces the icon to be rendered with a squared area, this helps icons alignment */
  squared?: boolean,
};

const Icon = styled(({ name, squared = true, ...props }: Props) => (
  <i {...props}>{name}</i>
))`
  /* use !important to prevent issues with browser extensions that change fonts */
  font-family: 'quid-icons' !important;
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: inherit;

  /* Enable Ligatures */
  letter-spacing: 0;
  font-feature-settings: 'liga';
  font-variant-ligatures: discretionary-ligatures;

  /* Better Font Rendering */
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;

  ${props =>
    props.squared &&
    css`
      display: inline-flex;
      align-items: center;
      justify-content: center;
      height: 1em;
      width: 1em;
    `};
`;

// @component
export default Icon;
