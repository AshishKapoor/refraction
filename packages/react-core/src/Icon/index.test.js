// @flow
import * as React from 'react';
import { mount } from 'enzyme';
import Icon from '.';

it('renders the expected markup', () => {
  const wrapper = mount(<Icon name="network" />);

  expect(wrapper).toMatchSnapshot();
});

it('renders a non squared Icon', () => {
  const wrapper = mount(<Icon name="network" squared={false} />);

  expect(wrapper).toMatchSnapshot();
});
