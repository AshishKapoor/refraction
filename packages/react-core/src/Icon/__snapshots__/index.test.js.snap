// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a non squared Icon 1`] = `
.emotion-0 {
  font-family: 'quid-icons' !important;
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: inherit;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-feature-settings: 'liga';
  font-variant-ligatures: discretionary-ligatures;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<Icon
  name="network"
  squared={false}
>
  <i
    className="emotion-0 emotion-1"
  >
    network
  </i>
</Icon>
`;

exports[`renders the expected markup 1`] = `
.emotion-0 {
  font-family: 'quid-icons' !important;
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: inherit;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-feature-settings: 'liga';
  font-variant-ligatures: discretionary-ligatures;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<Icon
  name="network"
>
  <i
    className="emotion-0 emotion-1"
  >
    network
  </i>
</Icon>
`;
